test/e2e/Makefile

```diff
@@ -0,0 +1,138 @@
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-kcp-remediation > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-kcp-remediation.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-md-remediation > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-md-remediation.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-zone > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-zone.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-account > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-account.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-domain > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-domain.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-template > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-template.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-cp-offering > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-cp-offering.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-worker-offering > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-invalid-worker-offering.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-insufficient-compute-resources > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-insufficient-compute-resources.yaml
+	$(KUSTOMIZE) build --load-restrictor LoadRestrictionsNone $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-node-drain > $(CLOUDSTACK_TEMPLATES)/v1beta1/cluster-template-node-drain.yaml
+
+## --------------------------------------
+## Docker build and push to local registry
+## --------------------------------------
+
+.PHONY: docker-build-and-push
+docker-build-and-push:
+	-d=$$(date +%s) \
+	&& IMG=$(IMG) $(MAKE) -C $(REPO_ROOT) docker-build \
+	&& IMG=$(IMG) $(MAKE) -C $(REPO_ROOT) docker-push \
+	&& echo "This step took $$(($$(date +%s)-d)) seconds"
+
+
+## --------------------------------------
+## End-to-End Tests
+## --------------------------------------
+
+.PHONY: e2e-essentials
+e2e-essentials: $(GINKGO) kind-cluster manifests cluster-templates docker-build-and-push
+
+.PHONY: run-e2e
+run-e2e: e2e-essentials ## Run End-to-End Tests
+	-d=$$(date +%s) \
+	&& PATH=$(BIN_DIR):${PATH} $(GINKGO) -v -trace -tags=e2e -skip=Conformance -nodes=1 --noColor=false ./... -- \
```

Comment:
PATH can be set via export, and we actually do that from the main Makefile.

So, if you use the main Makefile to call this one, you shouldn't even need to set this path var.

`$(GINKGO)` in particular should just be `gingko` since it's now in `$PATH`